lib/cwa/generateCwaLink.ts

```diff
@@ -0,0 +1,68 @@
+import { Checkin } from '../db'
+import {
+  CWALocationData,
+  QRCodePayload,
+  TraceLocationType,
+} from './trace_location'
+
+const base64toUint8 = (str: string): Uint8Array => {
+  const len = str.length
+  const bytes = new Uint8Array(len)
+  for (let i = 0; i < len; i++) {
+    bytes[i] = str.charCodeAt(i)
+  }
+  return bytes
+}
+
+const locationType = (locationType: string) => {
+  switch (locationType) {
+    case 'retail':
+      return TraceLocationType.LOCATION_TYPE_PERMANENT_RETAIL
+    case 'food_service':
+      return TraceLocationType.LOCATION_TYPE_PERMANENT_FOOD_SERVICE
+    case 'craft':
+      return TraceLocationType.LOCATION_TYPE_PERMANENT_CRAFT
+    case 'workplace':
+      return TraceLocationType.LOCATION_TYPE_PERMANENT_WORKPLACE
+    case 'educational_institution':
+      return TraceLocationType.LOCATION_TYPE_PERMANENT_EDUCATIONAL_INSTITUTION
+    case 'public_building':
+      return TraceLocationType.LOCATION_TYPE_PERMANENT_PUBLIC_BUILDING
+    case 'other':
+      return TraceLocationType.LOCATION_TYPE_PERMANENT_OTHER
+    default:
+      return TraceLocationType.LOCATION_TYPE_UNSPECIFIED
+  }
+}
+
+export const generateCwaLink = (checkin: Checkin) => {
+  const public_key = window.atob(
+    'gwLMzE153tQwAOf2MZoUXXfzWTdlSpfS99iZffmcmxOG9njSK4RTimFOFwDh6t0Tyw8XR01ugDYjtuKwjjuK49Oh83FWct6XpefPi9Skjxvvz53i9gaMmUEc96pbtoaA'
```

Comment:
I asked myself where is this coming from. Perhaps it's good to note the source https://github.com/corona-warn-app/cwa-documentation/blob/master/event_registration.md

Reply:
Added a comment about it.